gps/vcs_repo.go

```diff
@@ -26,38 +26,11 @@ type ctxRepo interface {
 	//ping(context.Context) (bool, error)
 }
 
-func newCtxRepo(s vcs.Type, ustr, path string) (ctxRepo, error) {
-	r, err := getVCSRepo(s, ustr, path)
-	if err != nil {
-		// if vcs could not initialize the repo due to a local error
-		// then the local repo is in an incorrect state. Remove and
-		// treat it as a new not-yet-cloned repo.
-
-		// TODO(marwan-at-work): warn/give progress of the above comment.
-		os.RemoveAll(path)
```

Comment:
This was essentially inlined into each `maybeSource`, which allows them to directly create their respective `vcs.Repo`s, rather than switching on the type, at the cost of bit of duplication. IMHO it is clearer, especially when trying to unravel the chain of similarly named types we create.